File: apps/api/v3/logic/meta.rb
# apps/api/v3/logic/meta.rb
#
# frozen_string_literal: true

# V3 Meta Logic
#
# Native V3 implementations following pure REST conventions.
# HTTP status codes indicate success/error — no `success` field in responses.

module V3
  module Logic
    # @api System metadata endpoints for health checks, version info, and
    #   supported locale listings. These endpoints do not require
    #   authentication.
    module Meta
      SCHEMAS = {
        system_status: { response: 'systemStatus' },
        system_version: { response: 'systemVersion' },
        get_supported_locales: { response: 'supportedLocales' },
      }.freeze

      # Get Supported Locales
      #
      # @api Returns the list of supported locales and the default locale
      #   for the application.
      def self.get_supported_locales(_req, _res)
        supported_locales = OT.supported_locales.map(&:to_s)
        default_locale    = OT.default_locale
        {
          locales: supported_locales,
          default_locale: default_locale,
          locale: default_locale,
        }
      end

      # System Status
      #
      # @api Returns the current operational status of the system.
      def self.system_status(_req, _res)
        {
          status: :nominal,
          locale: OT.default_locale,
        }
      end

      # System Version
      #
      # @api Returns the current application version as an array of
      #   version components.
      def self.system_version(_req, _res)
        {
          version: OT::VERSION.to_a,
          locale: OT.default_locale,
        }
      end
    end
  end
end

File: src/schemas/api/v3/responses/meta.ts
// src/schemas/api/v3/responses/meta.ts
//
// V3 JSON wire-format schemas for meta endpoints (status, version, locales).
// V3 omits the `success` field — HTTP status codes indicate success/error.
// V2 includes `success: boolean` in responses; see v2/responses/meta.ts.

import { z } from 'zod';

export const systemStatusResponseSchema = z.object({
  status: z.string(),
  locale: z.string(),
});

export const systemVersionResponseSchema = z.object({
  version: z.array(z.union([z.string(), z.number()])),
  locale: z.string(),
});

export const supportedLocalesResponseSchema = z.object({
  locales: z.array(z.string()),
  default_locale: z.string(),
  locale: z.string(),
});

export type SystemStatusResponse = z.infer<typeof systemStatusResponseSchema>;
export type SystemVersionResponse = z.infer<typeof systemVersionResponseSchema>;
export type SupportedLocalesResponse = z.infer<typeof supportedLocalesResponseSchema>;
